src/components/FileTree.vue

```
@@ -249,29 +252,30 @@ export default {
             };
         },
 
-        revisionPopover(f) {
-            let action;
-
+        diffAction(f) {
             if (f.ids !== undefined) {
                 if (f.ids[0] == null) {
-                    action = 'added';
+                    return 'added';
                 } else if (f.ids[1] == null) {
-                    action = 'deleted';
+                    return 'deleted';
                 } else {
-                    action = 'changed';
+                    return 'changed';
                 }
             } else if (f.revision !== undefined) {
-                action = f.revision == null ? 'deleted' : 'changed';
-            }
-
-            const text = `This file was ${action} in the teacher's revision.`;
-
-            if (!this.diffMode) {
-                return `${text} Click here to see the diff.`;
+                if (f.revision == null) {
+                    return 'deleted';
+                } else {
+                    return 'changed';
+                }
             } else {
-                return text;
+                return 'unchanged';
```

`unchanged` is never used as key, so maybe this should throw an error or it should be added to `diffLabels`.